src/raftstore/store/peer.rs

```diff
@@ -1275,6 +1281,9 @@ impl Peer {
                 if lease_to_be_updated {
                     let propose_time = self.find_propose_time(entry.get_index(), entry.get_term());
                     if let Some(propose_time) = propose_time {
+                        ctx.raft_metrics.commit_log.observe(duration_to_sec(
```

Comment:
I think commit duration should be observed from the head instead of the tail.

Reply:
You are right, PTAL again.